zipkin-ui/js/component_ui/spanPanel.js

```diff
@@ -25,7 +25,12 @@ export default component(function spanPanel() {
         const maybeObject = anno[$this.data('key')];
         // In case someone is storing escaped json as an annotation value
         // TODO: this class is not testable at the moment
-        $this.text($.type(maybeObject) === 'object' ? JSON.stringify(maybeObject) : maybeObject);
+        const type = $.type(maybeObject);
```

Comment:
ps this part is for annotations (not binary annotion/tags). This change would make the UI timeline look weird, as single-line is ideal. I would revert this part.